### ISSUE TITLE
Unbound ids checked after type-checking

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#547]
In the compiler workflow, type-checking happens before unbound-id checking, which can lead to surprising errors because the type-checker assumes all identifiers are defined.  Example:

```
x :: NotBound = 5
Expected to find NotBound at definitions: line 1, column 13 , required by definitions: line 1, column 0 , but instead found Number 
```

**Comment by schanzer** [comment]
I know this is minor, but: shouldn't it be "Number"? :)

**Comment by jpolitz** [comment]
Updated the issue text to be clearer -- the misspelling was the point.

**Comment by jpolitz** [comment]
Fixed by 9a6ca6a2984ab6b95b5e97829c39b50d1258ee53
